### PR TITLE
[92] Create an enum for `property` in modify method for zones

### PR DESCRIPTION
**Pull request by sam-i-am012**
Addresses #92. All tests pass.

### CODE REVIEW

File: src/main/java/org/irods/util/ResourceProperty.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.irods.util;
+
+public enum ResourceProperty {
```
Comment (korydraughn) [inline src/main/java/org/irods/util/ResourceProperty.java:3]: Documentation?

**Review by korydraughn — COMMENTED** [review]
Looks good overall.

Had one comment.

Do the tests pass?

File: src/main/java/org/irods/util/ResourceProperty.java (hunk 1 above)
Comment (sam-i-am012) [reply src/main/java/org/irods/util/ResourceProperty.java:3]: I was going to add documentation with the other PR #94 and leave the issue opened until that got merged, but I can go ahead and do it now.

Comment (korydraughn) [reply src/main/java/org/irods/util/ResourceProperty.java:3]: If this enum is new, let's go ahead and add the docs.

Had the enum already existed, then handling the documentation in the other PR would be more acceptable.

Comment (sam-i-am012) [reply src/main/java/org/irods/util/ResourceProperty.java:10]: Okok makes sense.

**Comment by sam-i-am012** [comment]
All tests pass and I've added documentation.

File: src/main/java/org/irods/util/ResourceProperty.java [hunk 2]
```diff
@@ -1,5 +1,12 @@
 package org.irods.util;
 
+/**
+ * Enum representing various properties for the iRODS system.
+ * <p>
+ *     This enum is used to ensure that only valid {@code property} values are passed to the modify() method used in
```
Comment (korydraughn) [inline src/main/java/org/irods/util/ResourceProperty.java:6]: Is `{@code property}` correct?

File: src/main/java/org/irods/util/ResourceProperty.java [hunk 3]
```diff
@@ -13,11 +20,21 @@ public enum ResourceProperty {
...
+     * @param value The string representation of the user type.
+     */
+    ResourceProperty(String value) {
+        this.value = value;
     }
 
+    /**
+     * Returns the string representation of this property.
+     *
+     * @return The string representation of the property.
```
Comment (korydraughn) [inline src/main/java/org/irods/util/ResourceProperty.java:35]: Probably don't need the bottom two lines because the main description covers it.

Comment (sam-i-am012) [reply src/main/java/org/irods/util/ResourceProperty.java]: Done, I'll do the same for the other enums.

File: src/main/java/org/irods/util/ResourceProperty.java (hunk 2 above)
Comment (sam-i-am012) [reply src/main/java/org/irods/util/ResourceProperty.java:6]: The enum is for the name of the property to modify

Comment (korydraughn) [reply src/main/java/org/irods/util/ResourceProperty.java:6]: I was misunderstanding what `{@code}` was doing.

Carry on.

**Review by korydraughn — COMMENTED** [review]
Squash it.

**Comment by sam-i-am012** [comment]
Done.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by sam-i-am012** [comment]
Done.

**Comment by korydraughn** [comment]
Merging.

**Pull request merged**